source/connection-string/tests/valid-warnings.yml

```diff
@@ -73,3 +73,15 @@ tests:
                 port: ~
         auth: ~
         options: ~
+    -
+        description: Comma in a key value pair causes a warning
+        uri: mongodb://example.com?authMechanismProperties=TOKEN_RESOURCE:mongodb://host1%2Chost2
```

Comment:
Is this really a warning? If we are decoding the entire option value, which is the present (though perhaps undocumented) behaviour for at least some drivers, then we get:

`TOKEN_RESOURCE:mongodb://host1,host2`

Which would contain two properties: `TOKEN_RESOURCE`, and `host2`, with the latter not having a `:`, or value assigned, which seems like it would be an error.

Reply:
I based this on "Any invalid Values for a given key MUST be ignored and MUST log a WARN level message"